sdks/python/apache_beam/options/pipeline_options.py

```diff
@@ -219,6 +231,11 @@ def __init__(self, flags=None, **kwargs):
         self._all_options[option_name] = getattr(
             self._visible_options, option_name)
 
+    # Override flags dependent on each other
+    for option_name, option_value in _PARSE_OPTIONS_WITH_PREFIX_NO.items():
+      if option_name in self._all_options:
+        self._all_options[option_value] = (not self._all_options[option_name])
```

Comment:
I think we should only expect inputs where `self._all_options[option_name] == True`, and print a warning that the False value is ignored otherwise. Then, we can assign `False` to the dest. This would be consistent with command-line parsing